deepeval/synthesizer/context_generator.py

```diff
@@ -2,120 +2,151 @@
+        print("finish generating contexts: ", contexts)
         return contexts, source_files
-    
-    
+
     ############### Load Docs #############################
     def _load_docs(self) -> List[Chunk]:
 
         def process_document(path):
-            doc_chunker = DocumentChunker(self.embedder, self.chunk_size, self.chunk_overlap)
+            doc_chunker = DocumentChunker(
+                self.embedder, self.chunk_size, self.chunk_overlap
+            )
             return doc_chunker.load_doc(path)
-        
+
         combined_chunks = []
 
         if not self.multithreading:
             for path in self.document_paths:
-                doc_chunker = DocumentChunker(self.embedder, self.chunk_size, self.chunk_overlap)
+                doc_chunker = DocumentChunker(
+                    self.embedder, self.chunk_size, self.chunk_overlap
+                )
                 chunks = doc_chunker.load_doc(path)
                 combined_chunks.extend(chunks)
         else:
             with ThreadPoolExecutor() as executor:
-                future_to_path = {executor.submit(process_document, path): path for path in self.document_paths}
+                future_to_path = {
+                    executor.submit(process_document, path): path
+                    for path in self.document_paths
+                }
                 for future in as_completed(future_to_path):
                     path = future_to_path[future]
                     try:
                         chunks = future.result()
                         combined_chunks.extend(chunks)
                     except Exception as exc:
-                        print(f'{path} generated an exception: {exc}')
-
+                        print(f"{path} generated an exception: {exc}")
 
         return combined_chunks
-    
+
     ############### Search N Chunks ########################
-    def _get_n_random_clusters(self, n: int, cluster_size: int) -> List[List[Chunk]]:
+    def _get_n_random_clusters(
+        self, n: int, cluster_size: int
+    ) -> List[List[Chunk]]:
         chunk_cluster = []
+        print("get n random chunks, cluster size: ", cluster_size)
+        print("n: ", n)
         random_chunks = self._get_n_random_chunks(n)
+        print(len(random_chunks))
         for chunk in random_chunks:
+            print("ok")
             cluster = [chunk]
-            cluster.extend(self._get_n_other_similar_chunks(chunk, n = cluster_size - 1))
+            cluster.extend(
+                self._get_n_other_similar_chunks(chunk, n=cluster_size - 1)
+            )
             chunk_cluster.append(cluster)
+        print(chunk_cluster)
         return chunk_cluster
 
     def _get_n_random_chunks(self, n: int) -> List[str]:
-        if not self.combined_chunks or len(self.combined_chunks) < n:
-            raise ValueError("Not enough chunks to return the requested number of random nodes.")
+        n = min(len(self.combined_chunks), n)
```

Comment:
Same here, confining it instead of raising a bug. I don't think here is where it is causing the extremely short context lengths.